controllers/external/tracker.go

```diff
@@ -44,13 +44,8 @@ type ObjectTracker struct {
 
 // Watch uses the controller to issue a Watch only if the object hasn't been seen before.
 func (o *ObjectTracker) Watch(log logr.Logger, obj client.Object, handler handler.EventHandler, p ...predicate.Predicate) error {
-	// Consider this a no-op if the controller isn't present.
-	if o.Controller == nil {
-		return nil
-	}
-
 	if o.Cache == nil || o.Scheme == nil {
```

Comment:
```suggestion
	if o.Cache == nil || o.Scheme == nil || o.Controller == nil {
```

Should it be like that?

Reply:
Yes, Its supposed to be like that, I just had it in different if condition, Thought of combining them, But did half job. I will update.

Reply:
Updated as required.